src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/SlimefunItem.java

```diff
@@ -50,6 +50,7 @@
 	private boolean replacing = false;
 	private boolean addon = false;
 	private String permission = "";
+	private String[] permissionMessage = {"&4&lLOCKED", "", "&rYou do not have Permission", "&rto access this Item"};
```

Comment:
This value is read-only atm, keeping it in RAM for all the thousands of Slimefunitems makes zero sense in your design, you should write to it in the register() block were all the other values are set as well.

Reply:
Would this work?